### PR TITLE
Add MedicationDispense Resource

### PR DESCRIPTION
**Pull request by kivarsson-va**
Lasershark team adding MedicationDispense Resource as per FHIR guidelines. Please let us know if there are any issues!

GH Issue: https://github.com/department-of-veterans-affairs/lasershark/issues/73

### CODE REVIEW

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+  @Valid List<Resource> contained;
+
+  @Valid List<Extension> extension;
+
+  @Valid List<Extension> modifierExtension;
+
+  // Medication Dispense
+  @Valid List<Identifier> identifier;
+
```
Comment (bschofield-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java:83]: Missing `partOf`

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  @Valid @NotNull Status status; // required
+
+  @Valid CodeableConcept statusReasonCodeableConcept;
+
+  @Valid Reference statusReasonReference;
+
+  @Valid List<CodeableConcept> category;
```
Comment (bschofield-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java:90]: not a list

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+  @Valid CodeableConcept medicationCodeableConcept;
+
+  @Valid Reference medicationReference;
+
+  @Valid @NotNull Reference subject;
```
Comment (bschofield-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java:96]: subject is not required

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  @Schema(name = "MedicationDispensePerformer")
+  public static class Performer implements AsList<Performer>, BackboneElement {
+    @Pattern(regexp = Fhir.ID)
+    String id;
+
+    @Valid List<Extension> extension;
+    @Valid List<Extension> modifierExtension;
+
+    @Valid CodeableConcept function;
+    @Valid @NotNull Reference actor; // required
```
Comment (bschofield-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java:163]: redundant `// required` comment

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  @Schema(name = "MedicationDispenseSubstitution")
+  public static class Substitution implements AsList<Substitution>, BackboneElement {
+    @Pattern(regexp = Fhir.ID)
+    String id;
+
+    @Valid List<Extension> extension;
+
+    @Valid List<Extension> modifierExtension;
+
+    @NotNull Boolean wasSubstituted; //required
```
Comment (bschofield-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java:244]: redundant `// required` comment

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java (hunk 1 above)
Comment (kivarsson-va) [reply us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/MedicationDispense.java:83]: where does `partOf` come from? i'm not seeing it in the FHIR spec

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**